### PR TITLE
feat(RichTextEditor): lazy load quill

### PR DESCRIPTION
**Pull request by OndrejTuma**
[FX-2759]

### Description

`QuillEditor` component is lazy loaded so that instance of quill does not load when using Picasso without tree shaking.
This way will the bundle size on other project will decrease and it should work out of the box for frameworks with SSR when importing Picasso components from the root like this:

```js
import { Container } from '@toptal/picasso'
```

### How to test

First of all you can [check temploy](https://picasso.toptal.net/FX-2759-quill-load/?path=/story/components-richtexteditor--richtexteditor) that RichTextEditor is working as expected. Then you can check whether quill is imported by following the next steps:

- Create a local project in Next.js using [Picasso guide](https://picasso.toptal.net/?path=/story/tutorials-how-to-use-server-side-rendering--how-to-use-server-side-rendering) - look for **Next.js** section.
- Don't stub quill (last sentence in step 5.)
- import (one) Picasso component without tree-shaking in `pages/index.tsx`: 
```js
import { Container } from '@toptal/picasso'
```
- run `yarn dev` and open `http://localhost:3000/`

You should see an error indicating that quill has been imported and parsed on the server

image

- now update Picasso version in package.json to alpha: `25.1.1-alpha-FX-2759-quill-load.0`
- run `yarn dev` and open `http://localhost:3000/`

Now it should work without an error
 
### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2759]: https://toptal-core.atlassian.net/browse/FX-2759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

**Comment by OndrejTuma** [comment]
@toptal-bot run package:alpha-release

File: cypress/component/PageSidebar.spec.tsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ describe('Sidebar', () => {
       mount(<DefaultExample collapsible />)
 
       // Make sure the sidebar is not hovered
-      cy.get('body').trigger('mousemove', { clientX: 0, clientY: 0 })
+      cy.get('body').trigger('mousemove', { clientX: 250, clientY: 0 })
```
Comment (OndrejTuma) [inline cypress/component/PageSidebar.spec.tsx:163]: I added this just to remove [flakiness](https://github.com/toptal/picasso/runs/6495920979?check_suite_focus=true) of the PageSidebar test. If it was fixed somewhere else or can be fixed more elegantly, I'm open to suggestions

Comment (TomasSlama) [reply cypress/component/PageSidebar.spec.tsx:163]: @OndrejTuma https://github.com/toptal/picasso/pull/2783 It was in draft moment ago, but I have just switched it ready for review

**Review by TomasSlama — APPROVED** [review]
I don't like the name of the `QuillEditorOutput`, but I'm not sure even if I like my suggestion 😂, maybe someone else will have a better idea of how to name it. Otherwise, it looks good. Good job 💪

**Comment by denieler** [comment]
@OndrejTuma I've tested on Davinci app and it looks like it doesn't work to me - 

The code I've used:
```
import { Form } from '@toptal/picasso-forms'
...
      <div>
        <Form onSubmit={() => {}}>
          Some Form without RichTextEditor
        </Form>
      </div>
```

and this is the bundle result - 

Screenshot 2022-05-19 at 17 34 01

File: packages/picasso/src/QuillEditor/QuillEditor.tsx [hunk 2]
```diff
@@ -77,18 +70,7 @@ const QuillEditor = forwardRef<HTMLDivElement, Props>(function QuillEditor(
...
-      variant='body'
-      color='dark-grey'
-      size='medium'
-      className={classes.root}
-      data-testid={dataTestId}
-      id={id}
-      ref={editorRef}
-    />
-  )
+  return <QuillEditorOutput ref={editorRef} data-testid={dataTestId} id={id} />
```
[the remark this answers is not in the document]
Comment (OndrejTuma) [reply packages/picasso/src/QuillEditor/QuillEditor.tsx:73]: That sounds better 👍 I'll adjust it

File: cypress/component/PageSidebar.spec.tsx (hunk 1 above)
Comment (OndrejTuma) [reply cypress/component/PageSidebar.spec.tsx:163]: Thanks! It seems like this change didnt help 🤔

**Comment by OndrejTuma** [comment]
> @OndrejTuma I've tested on Davinci app and it looks like it doesn't work to me

I tested it on bootstrapped davinci app - the separate chunk is generated but loaded in the app only when `Form.RichTextEditor` or `RichTextEditor` is mounted as a component. We should be fine 🤞

File: packages/picasso/src/QuillEditor/QuillEditor.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (OndrejTuma) [reply packages/picasso/src/QuillEditor/QuillEditor.tsx]: It's done @TomasSlama 🙇

**Comment by denieler** [comment]
> > @OndrejTuma I've tested on Davinci app and it looks like it doesn't work to me
> 
> I tested it on bootstrapped davinci app - the separate chunk is generated when `@toptal/picasso-forms` is used, but loaded in the app only when `Form.RichTextEditor` is mounted as a component. We should be fine 🤞

yep, my mistake, that was a generated chunk, but not used by the app when `RichTextEditor` has not been used in the code. Tested again, looks like working fine 👍

**Review by denieler — APPROVED** [review]

**Review by ozgurkececioglu — APPROVED** [review]
Didn't test but it looks really clean. GJ 👍

**Review by separatio — APPROVED** [review]

**Pull request merged**